File: proyecto-adoptapet/adoptapet-backend/src/main/java/com/proyecto/adoptapet/model/Adoptante.java
package com.proyecto.adoptapet.model;

import java.sql.Date;

import jakarta.persistence.Entity;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true) // Para que Lombok no ignore los campos de Usuario
public class Adoptante extends Usuario{
	private String nom_adoptante;
	
	private String ape_adoptante;
	
	private String dni;
	
	private Date fec_nacimiento;
	
	private String email;
	
	private String telefono;
	
	private String direccion;
}

File: ProyectoAndroid/app/src/main/java/com/cibertec/adoptapet/models/Adoptante.kt
package com.cibertec.adoptapet.models

data class Adoptante(
    val username: String,
    val password: String,
    val rol: String = "ROLE_ADOPTANTE",
    val nom_adoptante: String,
    val ape_adoptante: String,
    val dni: String,
    val fec_nacimiento: String,
    val email: String,
    val telefono: String,
    val direccion: String
)
